plugins/account_history/plugin.cpp

```diff
@@ -135,242 +157,290 @@ if( options.count(name) ) { \
+            op.get_required_owner_authorities(impd);
+            for (auto i : impd) {
+                impacted.insert(make_pair(i, operation_direction_type::any));
+            }
         }
 
         void operator()(const account_create_operation& op) {
-            impacted.insert(op.new_account_name);
-            impacted.insert(op.creator);
+            impacted.insert(make_pair(op.new_account_name, operation_direction_type::receiver));
+            impacted.insert(make_pair(op.creator, operation_direction_type::sender));
         }
 
         void operator()(const account_create_with_delegation_operation& op) {
-            impacted.insert(op.new_account_name);
-            impacted.insert(op.creator);
+            impacted.insert(make_pair(op.new_account_name, operation_direction_type::receiver));
+            impacted.insert(make_pair(op.creator, operation_direction_type::sender));
         }
 
         void operator()(const account_update_operation& op) {
-            impacted.insert(op.account);
+            impacted.insert(make_pair(op.account, operation_direction_type::any));
         }
 
         void operator()(const account_metadata_operation& op) {
-            impacted.insert(op.account);
+            impacted.insert(make_pair(op.account, operation_direction_type::any));
         }
 
         void operator()(const comment_operation& op) {
-            impacted.insert(op.author);
-            if (op.parent_author.size()) {
-                impacted.insert(op.parent_author);
+            if (op.author == op.parent_author) {
+                impacted.insert(make_pair(op.author, operation_direction_type::any));
+            } else {
+                if (op.parent_author.size()) {
+                    impacted.insert(make_pair(op.author, operation_direction_type::receiver));
+                } else {
+                    impacted.insert(make_pair(op.author, operation_direction_type::sender));
+                }
             }
         }
 
         void operator()(const delete_comment_operation& op) {
-            impacted.insert(op.author);
+            impacted.insert(make_pair(op.author, operation_direction_type::any));
         }
 
         void operator()(const vote_operation& op) {
-            impacted.insert(op.voter);
-            impacted.insert(op.author);
+            if (op.voter == op.author) {
+                impacted.insert(make_pair(op.voter, operation_direction_type::any));
+                impacted.insert(make_pair(op.author, operation_direction_type::any));
+            } else {
+                impacted.insert(make_pair(op.voter, operation_direction_type::sender));
+                impacted.insert(make_pair(op.author, operation_direction_type::receiver));
+            }
         }
 
         void operator()(const author_reward_operation& op) {
-            impacted.insert(op.author);
+            impacted.insert(make_pair(op.author, operation_direction_type::receiver));
```

Comment:
Rewards (and virtual ops) are tricky. There can be several cases:
1. Author is receiver because blockchain sends rewards him.
2. Author is sender because he created post and that action results in reward.
3. Author is both sender and receiver.

From usability perspective, it's better to set `any`, so api client can get all rewards using same direction. In case when he don't need `receiver` actions (votes spam on some post), rewards can be useful. What do you think?

Reply:
Author can't be sender. 
He create a post,  he vote the post, but he doesn't send reward to himself.
The blockchain check many factors and pay reward to the author.
Author is only receiver in this case.

Reply:
Yes, it doesn't fit good in terms sender/receiver. The initial "direction" idea was not about funds transfers, but about active/passive side of operation (e.g. to avoid votes casted to account, coz it's spammy).
In this case, author can be active side, because he initiated reward by post creation. Or can be passive, if we decide, that blockchain is active side here. But in 2nd case, withdrawal virtual operation receiver or interest reward receiver can be considered passive too.

Ok, let it be `receiver` if we are in sender/receiver world now.